src/System.Linq/tests/ToHashSetTests.cs

```diff
@@ -70,5 +70,363 @@ public void ThrowOnNullSource()
         {
             Assert.Throws<ArgumentNullException>(() => ((IEnumerable<object>)null).ToHashSet());
         }
+
+        [Fact]
+        public void WithConcat()
+        {
+            var first = Enumerable.Range(0, 50);
+            var second = Enumerable.Range(20, 50);
+
+            HashSet<int> result = first.Concat(second).ToHashSet();
+
+            Assert.NotNull(result);
+            Assert.Equal(70, result.Count);
```

Comment:
@jamesqo I've structured the tests using your idea of Theory, thanks for the suggestion.

I combined sources, comparers and actions, **ending up in 561 tests**, maybe this can be a concern, because some of these tests are checking something that already checked.

BTW, for simplicity to add new scenarios, I kept this way.

What do you think?